filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -726,7 +734,30 @@ void VulkanDriver::cancelExternalImage(void* image) {
 }
 
 bool VulkanDriver::getTimerQueryValue(Handle<HwTimerQuery> tqh, uint64_t* elapsedTime) {
-    return false;
+    VulkanTimerQuery* vtq = handle_cast<VulkanTimerQuery>(mHandleMap, tqh);
```

Comment:
I think there might be a race condition here. Because `getTimerQueryValue` is synchronous, it could be called on the main thread before `createTimerQueryR` has had a chance to construct the `VulkanTimerQuery` object. The Metal driver had to take scenarios like this into account.

At the very least I don't think we can trust that `vtq->startingQueryIndex` is necessarily valid here.

Reply:
Oh right, let me fix this up.

Reply:
I've added a commit, I took an approach similar to what you did.